### PR TITLE
Added ability to specify an existing Configmap for logstash pipelines

### PR DESCRIPTION
**Pull request by bdun1013**
Added ability to specify an existing Configmap for logstash pipelines

Signed-off-by: Brian Dunnigan <bdun1013dev@gmail.com>

- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [x] README.md updated with any new values or changes
- [x] Updated template tests in `${CHART}/tests/*.py` 
- [] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

I would like to be able to specify an existing configmap that contains my logstash pipelines, instead of using the one created in the chart with the logstashPipeline value. This will allow me to have templated pipeline configmaps for deployment that I can point logstash installations to.

One issue that might arise is that if the existing configmap is changed, the logstash pods will need to be manually restarted. The kubernetes community may soon begin an effort to start adding comfigmap rollouts in the new year: https://github.com/kubernetes/kubernetes/issues/22368#issuecomment-566785847. But for now there are a few approaches to handle automatic rollouts:

-  Create a new configmap instead of editing the existing one, point the logstash release to the new configmap, then run a helm upgrade

- Deploy the reloader [Reloader](https://github.com/stakater/Reloader) Kubernetes controller with the [Stable Helm Chart](https://github.com/helm/charts/tree/master/stable/reloader) and annotate deployments/statefulsets/daemonsets with annotations

The same can also be done for the configs (esConfig, kibanaConfig, etc) across the other charts and would be a useful feature. I will do those if this feature is accepted.

### CODE REVIEW

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
Hi @bpdunni, thank you for this PR and sorry for the delay.
This PR will need some time to review.

**Comment by qqshfox** [comment]
Should also change this
```
          {{- range $path, $config := .Values.logstashPipeline }}
          - name: logstashpipeline
            mountPath: /usr/share/logstash/pipeline/{{ $path }}
            subPath: {{ $path }}
          {{- end -}}
```

to

```
{{- if .Values.existingPipelineConfigmap }}
- name: logstashpipeline
  mountPath: /usr/share/logstash/pipeline
{{- else }}
{{- range $path, $config := .Values.logstashPipeline }}
- name: logstashpipeline
  mountPath: /usr/share/logstash/pipeline/{{ $path }}
  subPath: {{ $path }}
{{- end -}}
{{- end -}}
```

to mount existing ConfigMaps to pod.

**Comment by jmlrt** [comment]
still valid

**Comment by jmlrt** [comment]
jenkins test this please

File: logstash/tests/logstash_test.py [hunk 1]
```diff
@@ -622,6 +622,21 @@ def test_adding_in_pipeline():
...
+"""
+    r = helm_template(config)
+
+    s = r["statefulset"][name]["spec"]["template"]["spec"]
+
+    assert {"configMap": {"name": "uptime-pipeline"}, "name": "logstashpipeline"} in s[
+        "volumes"
+    ]
+
+    assert r["statefulset"][name]["spec"]["template"]["metadata"]["annotations"] == None
```
Comment (jmlrt) [inline logstash/tests/logstash_test.py:637]: This assertion doesn't test the pipeline from configmap feature and so should be removed

File: logstash/README.md [hunk 2]
```diff
@@ -98,54 +98,55 @@ useful for the [http input plugin][], for instance.
...
+| `antiAffinity`              | Setting this to hard enforces the [anti-affinity][] rules. If it is set to soft it will be done "best effort". Other values will be ignored                                                                                          | `hard`                                |
+| `envFrom`                   | Templatable string to be passed to the [environment from variables][] which will be appended to the `envFrom:` definition for the container                                                                                          | `[]`                                  |
+| `existingPipelineConfigmap` | Allows you to specify an existing configmap containing pipeline files to add to `/usr/share/logstash/pipeline/`. If set `logstashPipeline` is ignored.                                                                               | `""`                                  |
```
Comment (jmlrt) [inline logstash/README.md:106]: Maybe `externalPipelineConfigmap` would be more relevant.

**Review by jmlrt — CHANGES_REQUESTED** [review]
Hi @bdun1013, thanks for this PR and sorry for the delay.

I finally took the time to think about this PR and review it, and I think this would make sense for Logstash chart.

As mentioned in https://github.com/elastic/helm-charts/pull/450#issuecomment-656512803, the `volumeMounts` change is missing.

Also I would prefer naming this new value `externalPipelineConfigmap`.

**Pull request closed without merging**